### PR TITLE
feat: Adding Navigation Settings Functionality

### LINKED ISSUE #19163: [Task]: APIs for published and unpublished Navigation config
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Adding new database schemas 
- Update existing API for GET /pages (with added nav schema)
- Update existing API for PUT /applications/{applicationId} (which can update nav schema)
- Test Cases

### PR DESCRIPTION
**Pull request by NilanshBansal**
## Description
 
> This PR implements backend model and APIs for Part 1 of Improved App Viewer Navigation Project i.e implementing the navigation configuration settings 
> [Notion Doc](https://www.notion.so/appsmith/Improved-app-viewer-navigation-6126afa2a3694cad84382874a3f1bd9d#9d20423e53a54ec898083ab41f388af2)
> [Figma Design](https://www.figma.com/proto/EkPT7FdqmArkk9L7mySA5S/NAV---Designs?page-id=748%3A313757&node-id=935%3A478734&viewport=303%2C-336%2C0.04&scaling=min-zoom&starting-point-node-id=935%3A478734)

Fixes #19163 

## Implementation

It reuses the following existing APIs to implement storing and fetching navigation settings:

/api/v1/applications/{applicationId} (PUT)
/api/v1/pages?applicationId=\<applicationId>(GET)

Additional fields `unpublishedNavigationSetting` and `publishedNavigationSetting` have been added to the `application` collection.

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manual

**Update publishedNavigationSetting API**

- API Route: /api/v1/applications/{applicationId}
- API Method: PUT
- API JSON Response: [http://jsonblob.com/1057577523797573632](http://jsonblob.com/1057577523797573632)

image

**Update unpublishedNavigationSetting API**

- API Route: /api/v1/applications/{applicationId}
- API Method: PUT
- API JSON Response: [http://jsonblob.com/1057578112661078016](http://jsonblob.com/1057578112661078016)

image

**Database Record**

image

**GET Published and Unpublished Navigation Settings API**

- API Route: /api/v1/pages?applicationId=\<applicationId>
- API Method: GET
- API JSON Response: [http://jsonblob.com/1057580262598393856](http://jsonblob.com/1057580262598393856)

image

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java [hunk 1]
```diff
@@ -258,4 +264,25 @@ public static class EmbedSetting {
...
         private Boolean showNavigationBar;
     }
+
+
+    /**
+     * NavigationSetting stores the navigation configuration for the app
+     */
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
```
Comment (sharat87) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:274]: I'd encourage not adding these _until_ we need them somewhere. In this case, if we remove both of these annotations, the default no-args constructor will still be there. The all-args contructor, considering there's ten fields here, I don't think that constructor will be very readable/useful. 🙂

**Review by sharat87 — DISMISSED** [review]
Looks fine. Added a minor point.

**Comment by nidhi-nair** [comment]
Question more for my understanding, why did we decide against aggregating all unpublished and published fields in an app in one type?

Also, Nilansh, please use the [Fixes #foo](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) representation for attaching github issues to your PRs. Otherwise they will not autoclose on PR merge.

**Review by sharat87 — APPROVED** [review]

**Pull request merged**